test/compile/compile_tests.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+
+#
+# Creates a ctest that will succeed if the test fails to compile. Yep, you read that
+# correctly: SUCCESS == FAILURE for this set of tests. We use these to validate
+# compile-time asserts which guard against illegal template formation.
```

Comment:
This is so cool.

Reply:
Thanks but it does need refinement. I really want to find a better technique for this since you can't tell the difference between a syntax error and the static_assert actually working. 

One thing I could add is an post-processing step that EXPECTs certain compiler errors (i.e. the static_assert message). That would make this a robust technique. I'll add a task to do this.

Reply:
Added issue #224